### DISCUSSION TITLE
Switching network then executing transaction causes underlying network changed error

### DISCUSSION THREAD

**Discussion by neokry** [wevm/wagmi#243] [Q&A]
My dapp is mulitchain so having users switch chains then execute a transaction always causes this issue.

`Error: underlying network changed (event="changed", network={"name":"maticmum","chainId":80001,"ensAddress":null,"_defaultProvider":null}, detectedNetwork={"name":"rinkeby","chainId":4,"ensAddress":"0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e"}, code=NETWORK_ERROR, version=providers/5.5.3)`

This is the fix from the ethers side of things but not sure how to set this option with wagmi:
https://github.com/ethers-io/ethers.js/issues/1107

**Comment by SahilVasava** [comment]
Hey @neokry were you able to resolve this issue?
I am stuck on same thing. I am using `switchNetwork` to switch. So far I was able to switch the provider but the signer doesn't switch and therefor causes above error.

**Reply by 0xxmemo** [reply]
Hello man im having the same problem can you help me fix this please
